compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6202,6 +6205,23 @@ public void generateFieldsForUserUnspecifiedRecordFields(BLangRecordLiteral reco
         generateFieldsForUserUnspecifiedRecordFields(recordType, userSpecifiedFields, fieldNames, pos, isReadonly);
     }
 
+    private static boolean isSpreadingAnOpenRecord(List<RecordLiteralNode.RecordField> userSpecifiedFields) {
+        for (RecordLiteralNode.RecordField field : userSpecifiedFields) {
+            if (!(field instanceof BLangRecordLiteral.BLangRecordSpreadOperatorField spreadOperatorField)) {
+                continue;
+            }
+            BType type = Types.getReferredType(spreadOperatorField.expr.getBType());
+            if (!(type instanceof BRecordType recordType)) {
+                return true;
+            }
+            if (recordType.restFieldType != null &&
+                    Types.getImpliedType(recordType.restFieldType).tag != TypeTags.NEVER) {
```

Comment:
The more accurate check here is org.wso2.ballerinalang.compiler.semantics.analyzer.Types#isNeverTypeOrStructureTypeWithARequiredNeverMember(org.wso2.ballerinalang.compiler.semantics.model.types.BType).